xarray/tests/test_backends.py

```diff
@@ -1535,6 +1535,66 @@ def test_keep_chunksizes_if_no_original_shape(self) -> None:
                 ds["x"].encoding["chunksizes"], actual["x"].encoding["chunksizes"]
             )
 
+    def test_preferred_chunks_is_present(self) -> None:
+        ds = Dataset({"x": [1, 2, 3]})
+        chunksizes = (2,)
+        ds.variables["x"].encoding = {"chunksizes": chunksizes}
+
+        with self.roundtrip(ds) as actual:
+            assert actual["x"].encoding["preferred_chunks"] == {"x": 2}
+
+    @requires_dask
+    def test_auto_chunking_is_based_on_disk_chunk_sizes(self) -> None:
+        x_size = y_size = 1000
+        y_chunksize = y_size
+        x_chunksize = 10
+
+        with self.create_chunked_file((1, y_size, x_size), (1, y_chunksize, x_chunksize)) as tmp_file:
+            with dask.config.set({"array.chunk-size": "100KiB"}):
+                with open_dataset(tmp_file, chunks="auto", engine=self.engine) as ds:
+                    t_chunks, y_chunks, x_chunks = ds["image"].data.chunks
+                    assert all(np.asanyarray(y_chunks) == y_chunksize)
+                    # Check that the chunk size is a multiple of the file chunk size
+                    assert all(np.asanyarray(x_chunks) % x_chunksize == 0)
+
+    @requires_dask
+    def test_base_chunking_uses_disk_chunk_sizes(self) -> None:
+        x_size = y_size = 1000
+        y_chunksize = y_size
+        x_chunksize = 10
+
+        with self.create_chunked_file((1, y_size, x_size), (1, y_chunksize, x_chunksize)) as tmp_file:
+            with open_dataset(tmp_file, chunks={}, engine=self.engine) as ds:
+                for chunksizes, expected in zip(ds["image"].data.chunks, (1, y_chunksize, x_chunksize)):
+                    assert all(np.asanyarray(chunksizes) == expected)
+
+    @contextlib.contextmanager
+    def create_chunked_file(self, array_shape, chunk_sizes) -> None:
+        t_size, y_size, x_size = array_shape
+        t_chunksize, y_chunksize, x_chunksize = chunk_sizes
+
+        with create_tmp_file() as tmp_file:
+            with nc4.Dataset(tmp_file, mode="w") as nc:
+                nc.createDimension("t", t_size)
+                nc.createDimension("x", x_size)
+                nc.createDimension("y", y_size)
+                nc.createVariable("image", "int16", ("t", "y", "x"), fill_value=-1,
+                                  chunksizes=(t_chunksize, y_chunksize, x_chunksize))
+                v = nc.variables["image"]
+                v[:] = np.arange(t_size * x_size * y_size).reshape((t_size, y_size, x_size))
```

Comment:
While is this a fine way to create these test files, I would rather see us use xarray's backends here. The [`chunksizes`](https://docs.xarray.dev/en/stable/user-guide/io.html#chunk-based-compression) encoding option can be used to specify encoded chunksize in the netcdf4/h5netcdf engines.

Reply:
Good idea, done in 29c93b8